Vulkan-Hpp breaks with -Werror=undef on definition of assert

This appeared when compiling Vulkan-Hpp with the Android NDK. It appears that `assert` is not defined as a macro, but as a function, and this 
```
_deps/vulkanheaders-src/include/vulkan/vulkan.hpp:38:28: error: 'assert' is not defined, evaluates to 0 [-Werror,-Wundef]
   38 | #if ( VULKAN_HPP_ASSERT == assert )
      |                            ^
```

I'm not exactly sure what the right way to solve this is, or what exact the intention of this entire statement is. Surely if `assert` is already defined, then `<cassert>` is also already included?

Update: It seems to be critical here that I'm using the Vulkan headers via FetchContent, which causes them to be included using `-I`. When using `-isystem` as when included via regular find_package, there are no problems. Also note that I'm using `-Werror=undef`. I guess `isystem` causes the compiler to normally hide these problems.

--- a/snippets/includes.hpp
+++ b/snippets/includes.hpp
@@ -1,33 +1,33 @@
 #include <vulkan/vulkan_hpp_macros.hpp>
 
 #if defined( VULKAN_HPP_ENABLE_STD_MODULE ) && defined( VULKAN_HPP_STD_MODULE )
-#include <string.h>
+#  include <string.h>
 import VULKAN_HPP_STD_MODULE;
 #else
-# include <algorithm>
-# include <array>     // ArrayWrapperND
-# include <string.h>  // strnlen
-# include <string>    // std::string
-# include <utility>   // std::exchange
+#  include <algorithm>
+#  include <array>  // ArrayWrapperND
+#  include <cassert>
+#  include <string.h>  // strnlen
+#  include <string>    // std::string
+#  include <utility>   // std::exchange
+#  if 17 <= VULKAN_HPP_CPP_VERSION
+#    include <string_view>
+#  endif
+#  if !defined( VULKAN_HPP_DISABLE_ENHANCED_MODE )
+#    include <tuple>   // std::tie
+#    include <vector>  // std::vector
+#  endif
+#  if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
+#    include <compare>
+#  endif
+#  if !defined( VULKAN_HPP_NO_EXCEPTIONS )
+#    include <system_error>  // std::is_error_code_enum
+#  endif
+#  if defined( VULKAN_HPP_SUPPORT_SPAN )
+#    include <span>
+#  endif
 #endif
 #include <vulkan/${vulkan_h}>
-
-#if 17 <= VULKAN_HPP_CPP_VERSION && !( defined( VULKAN_HPP_ENABLE_STD_MODULE ) && defined( VULKAN_HPP_STD_MODULE ) )
-#  include <string_view>
-#endif
-
-#if !defined( VULKAN_HPP_DISABLE_ENHANCED_MODE ) && !( defined( VULKAN_HPP_ENABLE_STD_MODULE ) && defined( VULKAN_HPP_STD_MODULE ) )
-#  include <tuple>   // std::tie
-#  include <vector>  // std::vector
-#endif
-
-#if !defined( VULKAN_HPP_NO_EXCEPTIONS ) && !( defined( VULKAN_HPP_ENABLE_STD_MODULE ) && defined( VULKAN_HPP_STD_MODULE ) )
-#  include <system_error>  // std::is_error_code_enum
-#endif
-
-#if ( VULKAN_HPP_ASSERT == assert )
-#  include <cassert>
-#endif
 
 #if VULKAN_HPP_ENABLE_DYNAMIC_LOADER_TOOL == 1
 #  if defined( __unix__ ) || defined( __APPLE__ ) || defined( __QNX__ ) || defined( __Fuchsia__ )
@@ -43,12 +43,4 @@ extern "C" __declspec( dllimport ) HINSTANCE __stdcall LoadLibraryA( char const 
 extern "C" __declspec( dllimport ) int __stdcall FreeLibrary( HINSTANCE hLibModule );
 extern "C" __declspec( dllimport ) FARPROC __stdcall GetProcAddress( HINSTANCE hModule, const char * lpProcName );
 #  endif
-#endif
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR ) && !( defined( VULKAN_HPP_ENABLE_STD_MODULE ) && defined( VULKAN_HPP_STD_MODULE ) )
-#  include <compare>
-#endif
-
-#if defined( VULKAN_HPP_SUPPORT_SPAN ) && !( defined( VULKAN_HPP_ENABLE_STD_MODULE ) && defined( VULKAN_HPP_STD_MODULE ) )
-#  include <span>
 #endif

--- a/vulkan/vulkan.hpp
+++ b/vulkan/vulkan.hpp
@@ -15,29 +15,29 @@
 import VULKAN_HPP_STD_MODULE;
 #else
 #  include <algorithm>
-#  include <array>     // ArrayWrapperND
+#  include <array>  // ArrayWrapperND
+#  include <cassert>
 #  include <string.h>  // strnlen
 #  include <string>    // std::string
 #  include <utility>   // std::exchange
+#  if 17 <= VULKAN_HPP_CPP_VERSION
+#    include <string_view>
+#  endif
+#  if !defined( VULKAN_HPP_DISABLE_ENHANCED_MODE )
+#    include <tuple>   // std::tie
+#    include <vector>  // std::vector
+#  endif
+#  if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR )
+#    include <compare>
+#  endif
+#  if !defined( VULKAN_HPP_NO_EXCEPTIONS )
+#    include <system_error>  // std::is_error_code_enum
+#  endif
+#  if defined( VULKAN_HPP_SUPPORT_SPAN )
+#    include <span>
+#  endif
 #endif
 #include <vulkan/vulkan.h>
-
-#if 17 <= VULKAN_HPP_CPP_VERSION && !( defined( VULKAN_HPP_ENABLE_STD_MODULE ) && defined( VULKAN_HPP_STD_MODULE ) )
-#  include <string_view>
-#endif
-
-#if !defined( VULKAN_HPP_DISABLE_ENHANCED_MODE ) && !( defined( VULKAN_HPP_ENABLE_STD_MODULE ) && defined( VULKAN_HPP_STD_MODULE ) )
-#  include <tuple>   // std::tie
-#  include <vector>  // std::vector
-#endif
-
-#if !defined( VULKAN_HPP_NO_EXCEPTIONS ) && !( defined( VULKAN_HPP_ENABLE_STD_MODULE ) && defined( VULKAN_HPP_STD_MODULE ) )
-#  include <system_error>  // std::is_error_code_enum
-#endif
-
-#if ( VULKAN_HPP_ASSERT == assert )
-#  include <cassert>
-#endif
 
 #if VULKAN_HPP_ENABLE_DYNAMIC_LOADER_TOOL == 1
 #  if defined( __unix__ ) || defined( __APPLE__ ) || defined( __QNX__ ) || defined( __Fuchsia__ )
@@ -53,14 +53,6 @@ extern "C" __declspec( dllimport ) HINSTANCE __stdcall LoadLibraryA( char const 
 extern "C" __declspec( dllimport ) int __stdcall FreeLibrary( HINSTANCE hLibModule );
 extern "C" __declspec( dllimport ) FARPROC __stdcall GetProcAddress( HINSTANCE hModule, const char * lpProcName );
 #  endif
-#endif
-
-#if defined( VULKAN_HPP_HAS_SPACESHIP_OPERATOR ) && !( defined( VULKAN_HPP_ENABLE_STD_MODULE ) && defined( VULKAN_HPP_STD_MODULE ) )
-#  include <compare>
-#endif
-
-#if defined( VULKAN_HPP_SUPPORT_SPAN ) && !( defined( VULKAN_HPP_ENABLE_STD_MODULE ) && defined( VULKAN_HPP_STD_MODULE ) )
-#  include <span>
 #endif
 
 static_assert( VK_HEADER_VERSION == 312, "Wrong VK_HEADER_VERSION!" );

